llvm/lib/Transforms/PeepholeOptimizationCourse/Obfuscate8Bytes/Obfuscate8BytesAdd.cpp

```diff
@@ -0,0 +1,86 @@
+#include "llvm/Transforms/PeepholeOptimizationCourse/Obfuscate8BytesNewPassManager.h"
+
+#include "llvm/Pass.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/InstIterator.h"
+#include "llvm/IR/Constants.h"
+#include "llvm/IR/LegacyPassManager.h"
+#include "llvm/IR/InstrTypes.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/Support/raw_ostream.h"
+#include "llvm/Support/Debug.h"
+#include "llvm/Transforms/IPO/PassManagerBuilder.h"
+#include "llvm/Transforms/Utils/BasicBlockUtils.h"
+#define DEBUG_TYPE "obfuscate"
+
+using namespace llvm;
+namespace {
+
+	bool isAddWithI8(Instruction const* instruction) {
```

Comment:
How does that work? You declared `isAddWithI8` and `doObfuscateI8Add` function in the header for the pass with new manager, implemented them in source of pass with old manager and made definition inside anonymous namespace..

Reply:
Fixed in new commit